### PR TITLE
manage z-index/shadow

### PR DESCRIPTION
**Pull request by joekarasek**
I went through all the components and didn't see any visual regressions...
 
Thoughts?

### CODE REVIEW

**Comment by joekarasek** [comment]
The PL sticky component "view all" is a little funky. Looks like the sticky demo is being hidden by the sticky section header. I put the sticky demo on top, but it still looks terrible. Thoughts on how to better demonstrate this?

**Comment by joekarasek** [comment]
Changes have been implemented as per Salem's requests.
Note: The styling of 'Helpful Links' on the Sticky Docs page is handled in PR-495 https://github.com/bolt-design-system/bolt/pull/495

File: packages/core/styles/02-tools/tools-z-index/_tools-z-index.scss [hunk 1]
```diff
@@ -0,0 +1,18 @@
+/* ------------------------------------ *\
+  Z Index MIXIN
+\* ------------------------------------ */
+
+@mixin bolt-z-index($key: "content") {
+  @if map-has-key($bolt-z-index, $key) {
+    z-index: map-get($bolt-z-index, $key);
+  }
```
Comment (EvanLovely) [inline packages/core/styles/02-tools/tools-z-index/_tools-z-index.scss]: Add an `@else` that throws an `@error` here please

File: packages/core/styles/01-settings/_settings-z-index.scss [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  popover: 6 * $z-index-range,
+  nav: 5 * $z-index-range,
+  contentTop: 4 * $z-index-range,
+  content: 3 * $z-index-range,
+  contentBottom: 2 * $z-index-range,
+  backgroundTop: 1 * $z-index-range,
+  background: 0 * $z-index-range,
+  backgroundBottom: -1 * $z-index-range,
+);
+
```
Comment (EvanLovely) [inline packages/core/styles/01-settings/_settings-z-index.scss]: Remove `$z-index-range` implementation - even though it does create space between them, we still set `contentTop` to 4 and `nav` to 5 and so if we wanted to put something between them we'd use 4.5 which I don't like too much. The abstraction doesn't provide enough benefits.

File: apps/pattern-lab/src/_patterns/02-components/sticky/00-sticky-docs.twig [hunk 3]
```diff
@@ -14,4 +14,14 @@
...
   readmeFile: '@bolt-components-sticky/README.md',
-} only %}
\ No newline at end of file
+  links: [
+    {
+      url: '/pattern-lab/?p=components-sticky-simple-example',
+      text: 'Sticky Simple Example'
+    },
+    {
+      url: '/pattern-lab/?p=components-sticky-with-content-example',
```
Comment (EvanLovely) [inline apps/pattern-lab/src/_patterns/02-components/sticky/00-sticky-docs.twig]: It should be like this for both of the `url`s:

```diff
- url: '/pattern-lab/?p=components-sticky-with-content-example'
+ url: link['components-sticky-with-content-example'],
```

File: apps/pattern-lab/src/_patterns/01-styleguide/40-utilities/z-index.twig [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    {% if list|slice(0, -1)|length  %}
+      {{ _self.recursiveZindexDemo(list|slice(0, -1)) }}
+    {% endif %}
+    <p>Z-Index: <code>{{ list|keys|last }}</code>: {{ list|last }}</p>
+  </div>
+{% endmacro %}
+
+<h4>Z-index Values</h4>
+
+{{ _self.recursiveZindexDemo(bolt.data.zIndex) }}
```
Comment (EvanLovely) [inline apps/pattern-lab/src/_patterns/01-styleguide/40-utilities/z-index.twig]: Super slick! Nice work! 👍

**Comment by joekarasek** [comment]
@EvanLovely The changes you've requested have been made. Any further input?
@sghoweri Thoughts?

File: apps/pattern-lab/src/_patterns/02-components/sticky/00-sticky-docs.twig (hunk 3 above)
Comment (sghoweri) [reply apps/pattern-lab/src/_patterns/02-components/sticky/00-sticky-docs.twig]: @joekarasek @EvanLovely we still gonna update this? (I think we should - just double checking)

File: packages/core/styles/02-tools/tools-z-index/_tools-z-index.scss (hunk 1 above)
Comment (sghoweri) [reply packages/core/styles/02-tools/tools-z-index/_tools-z-index.scss]: @joekarasek think we might have missed this other one too - can we add the `@else` error handler to this real quick?

**Pull request merged**